In [29]:
wd = getwd()
data = read.csv(paste(wd, "/", "anes_pilot_2018.csv", sep=""))
head(data)


version                                     caseid weight    weight_spss form
1 ANES 2018 Pilot Study main version 20190129 1      1.2555080 0.7286969   1   
2 ANES 2018 Pilot Study main version 20190129 2      0.5694911 0.3305327   2   
3 ANES 2018 Pilot Study main version 20190129 3      0.9720616 0.5641846   1   
4 ANES 2018 Pilot Study main version 20190129 4      2.4500732 1.4220225   1   
5 ANES 2018 Pilot Study main version 20190129 5      1.6348200 0.9488495   2   
6 ANES 2018 Pilot Study main version 20190129 6      1.8184054 1.0554025   1   
  follow addtime reg whenreg howreg ... rand_emp rand_climate rand_rural
1 4      2       1    4      -1     ... 1        2            2         
2 1      2       3   -1      -1     ... 2        1            1         
3 4      3       3   -1      -1     ... 2        2            1         
4 4      2       1    4      -1     ... 2        2            1         
5 1      3       1    4      -1     ... 2        1            2         
6 3      2       2    2       2     ... 1        2            1         
  rand_pk rand_pid starttime           endtime             duration
1 1       1        12/10/2018 18:34:11 12/10/2018 20:00:03  5152   
2 0       2        12/10/2018 19:26:37 12/10/2018 19:53:12  1595   
3 1       2        12/10/2018 19:48:44 12/10/2018 20:21:21  1957   
4 0       1        12/10/2018 19:03:49 12/10/2018 19:21:08  1039   
5 1       2        12/10/2018 18:28:14 12/10/2018 18:43:23   909   
6 1       2        12/10/2018 23:58:17 12/11/2018 5:10:57  18760   
  optintimestamp completes
1 10/15/15        46      
2 7/16/16         55      
3 6/13/14          8      
4 10/28/15        42      
5 11/11/15        33      
6 5/9/15         185

In [25]:
nrow(data)


[1] 2500

In [26]:
ncol(data)


[1] 767

In [27]:
length(data)

[1] 767

Data context: https://electionstudies.org/wp-content/uploads/2019/02/anes_pilot_2018_userguidecodebook.pdf

# Assignment

1. 5 Questions <br/>
2. 4 different questions (primary focus) <br/>
3. 1 question to come up with on our own (secondary - will evolve with time) <br/>
4. Additional points to consider if we can crack sampling piece (3 points) - using the "weight variable"

## Common aspects of each question

1. **Introduce** - Explain how your variables are operationalized. Comment on any gaps that you can identify between your operational definitions and the concepts you are trying to study. <br/>
2. **EDA**- This should include a treatment of non-response and other special codes, basic sanity checks, and a justification for any values that are removed. Use visual tools to assess the relationship among your variables and comment on any features you find. <br/>
3. **Test setup** - Explain why your test is the most appropriate choice. List and evaluate all assumptions for your test. <br/>
4. **Result / Discussion** - Explain (1) the statistical significance of your result, and (2) the practical significance of your result. Make sure you relate your findings to the original research question






## Few concepts

**NON PROBABILITY SAMPLING**

The study was conducted on the Internet using the YouGov panel. The YouGov panel
consists of a large and diverse set of over a million respondents who have volunteered to
complete surveys online and who regularly receive invitations to do so. They receive points
usually worth about 21 to 50 cents for each survey they complete. The points are
redeemable for various gift cards. A respondent has to complete about 40 surveys to be
eligible for any reward. <br/>

This is voluntary sampling: https://www.youtube.com/watch?v=-kwdXEXC7yE (a form of non probability - non random sampling)

**DIDNT UNDERSTAND**: Respondents were selected from the YouGov panel by sample matching. **Matching is
intended to make the individuals who complete the survey represent the population on the
variables used for matching.** Respondents were matched to U.S. citizens in the 2016
American Community Survey (ACS) sample by gender, age, race, and education. **Matched
cases were weighted to the ACS frame using a propensity score model based on the prior
variables as well as geographic region.** After data collection the sample was weighted by
YouGov to match population characteristics for 2016 presidential candidate choice, gender,
age, race/ethnicity, and education.

**CONDITIONAL IGNORABILITY**

The basis for population inference: an assumption of conditional ignorability.
https://www.fragilefamilieschallenge.org/causal-inference/

Given the variables used in weighting and matching (X), survey participation (D) is independent of variables measured in the survey (Y).
The assumption is, in other words, that after controlling for the matching and weighting factors (gender, age, race/ethnicity, education, geographic region, and presidential candidate choice), participation or non-participation in the survey is not systematically related to any of the other survey questions.

Conditional ignorability: {Y(0),Y(1)} 丄 D | X

Given a set of covariates X, conditional ignorability states that treatment asignment D is independent of the potential outcomes that would be realized under treatment Y(1) and control Y(0). In other words, two observations with the same set of covariates X but with different treatment statuses can be compared to estimate the causal effect of the treatment for these observations.

**NET RESULT: If this assumption holds, a useful consequence is that the
sample distribution of survey variables will (at least approximately) match the distribution
of those variables in the population**

**NOTE ON STANDARD ERRORS**

CAVEAT: Standard errors are measures of variability in
sample statistics (estimates of population parameters) arising from random sampling. Because the YouGov survey data were
not generated by a probability sampling plan, **some practitioners recommend against using
the term “standard error”** (and related terms such as “sampling error,” “margin of error”
and “confidence intervals”) when reporting analyses of these data at least without noting
that these terms are not being used in their usual sense. **Valid inference of any “standard
error” reported for these data rests on the ignorability assumption noted above.**

**THE WEIGHT VARIABLE**

This variable is normed to have mean 1.0, such that the number of weighted observations
equals to the number of unweighted observations (2,500). The weight variable ranges
from 0.08 to 7.01 and has **variance 0.72**. Using this weight variable means that the
resulting standard errors and confidence intervals will be larger than those from analyses
that naively treat the data as a simple random sample (but SPSS users and other users of
frequency weights should see the note below on “SPSS weights”.) With a simple random
sample of 2,500 observations, a sample estimate of a proportion of .5 would have a
standard error of 0.01 and a 95% confidence interval of +/- 0.0196. **In this data set, the
weights results in a loss of precision relative to a simple random sample, by a factor
proportional to the variance of the weights (recall that under simple random sampling, any
weight variable would be constant and have variance zero)**; standard errors and confidence
intervals are inflated by a term proportional to the standard deviation of the weights

In the current dataset, the pseudo-design effect due to weighting is 1.723, and the root
design effect is 1.313. This means that, **provided that the ignorability assumption noted
above holds true**, the sample’s statistical power is equivalent to a simple random sample
size of 2,500/1.723 = 1,451, and **the expected “sampling error” (which is, again, not a
conventional sampling error because this is not a probability sample) for an estimated
proportion of .5 would be increased by a factor of 31.3% (0.0196 × 1.313) to .0257**

**Note that this 31.3% comes from 1.313**

**DATA COLLECTION**

YouGov invited 6,500 panelists to complete the survey. Of these,
3,322 started the questionnaire and 3,178 did not respond. Of those who started, 101 were
ineligible non-citizens and 311 did not answer the entire questionnaire (i.e., they are
considered breakoffs). The remaining 2,910 completed the questionnaire, of whom 131
were rejected by YouGov on quality-control grounds (such as completion time and item
nonresponse). Of the remaining 2,779, there were 2,500 selected (and 279 not selected) to
produce the final dataset by using the matching procedure described above. As described in
part 6 below, **the data from the 279 cases that were not selected are available on a separate
auxiliary data file.**

**DATA DESCRIPTION**

The variables from pid7 through race_other (positions 283 through 309) are YouGov
profile survey data. These data were collected on previously-completed questionnaires.


pk_germ and pk_cjus are scripted variables asking about different political knowledge related questions about Germany and US officials

The three variables pdl_inputzip, pdl_region, and pdl_cassregcountyname (positions
310-312) are geographic data for ZIP code, census region, and county. The ZIP and county
are restricted access data, not included on the public-use data file to protect respondent
confidentiality.

The variables from acaapprove_skp through turnout18w_skp (positions 313-499) are item
nonresponse prompt flags. When a respondent initially declined to answer a question, he or
she was asked to please answer. When this request was made, the occurrence was recorded.

The variables from follow_page_timing through honest_page_timing (positions 500-685)
show the time in seconds to complete the ANES questionnaire pages.

The variables from ord_gehope through ord_imrelieve (positions 686-750) record the
order in which indicated items or response options were displayed.

The variables from rand_votestop through rand_pid (positions 751-762) are
randomization variables referenced in the questionnaire specifications.

starttime, endtime, and duration, (positions 763-765) respectively indicate the date and
time that the respondent started the ANES questionnaire, the date and time the respondent
finished the questionnaire, and the duration of the interview in seconds.

The final variables on the file, optintimestamp (766) and completes (the last variable on
the file at position 767) respectively give the date the respondent joined the YouGov panel
and the total number of YouGov surveys previously completed, excluding the initial profile
survey. If optintimestamp is missing, this indicates that the respondent opted in to the
panel before March 2005, when opt-in date recording commenced.

## Initial Analysis

### Question 1: Do US voters have more respect for the police or for journalists?

Will need to use ftpolice (How would you rate the police?) and ftjournal (How would you rate journalists?)

**Follow up - what is ftpolice_therm and ftjournal_therm**

Possible other variables to look at: 
1. ftpolice_skp, ftpolice_page_timing, ord_ftpolice <br/>
2. ftjournal_skp, ftjournal_page_timing, ord_ftjournal

Will likeley need to do a "matched pairs sign test" https://www.youtube.com/watch?v=IcLSKko2tsg 

Note that these are ordinal / likert variables from 1-100  (think of it as a scale of agree / strongly agree etc.) - except that its way more granular.

Note that here we are being asked to simply measure a binary variable (have more / less respect), vs measure "how much more respect" does one group have over another.

We will calculate proportion of total US voters who have more respect for police P(police) = # (where ftpolice > ftjournal) / total

Hypothesis will be:

Ho: P(police) = 0.5 <br/>
Ha: P(police) != 0.5

In [30]:
summary(data$ftpolice)

   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
   0.00   47.00   70.00   64.68   90.00  100.00 

In [32]:
summary(data$ftjournal)

   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
  -7.00   21.00   52.00   52.26   82.00  100.00 

### Question 2: Are Republican voters older or younger than Democratic voters?

The variables from pid7 through race_other (positions 283 through 309) are YouGov profile survey data.


In [38]:
cols = seq(from = 283, to = 309, by = 1)
head(data[, cols])

pid7 birthyr gender educ marstat child18 employ employ_t faminc_new region
1 -4   1986    2      4    5       2       1      __NA__    3         1     
2 -4   1972    2      6    6       2       1      __NA__    4         3     
3 -4   1999    2      2    5       2       8      __NA__   97         1     
4 -4   1975    2      3    3       1       2      __NA__    3         3     
5 -4   1989    2      3    1       1       1      __NA__    4         1     
6 -4   1992    2      1    5       2       1      __NA__    2         3     
  ... pew_churatd pew_prayer religpew religpew_t religpew_protestant
1 ... 5           6           2       __NA__     98                 
2 ... 6           7          11       __NA__      7                 
3 ... 2           5           2       __NA__     98                 
4 ... 7           1           2       __NA__     98                 
5 ... 3           3           1       __NA__      1                 
6 ... 2           2           2       __NA__     98                 
  religpew_protestant_t immigrant inputstate race race_other
1 __NA__                5         42         1    __NA__    
2 __NA__                4         12         1    __NA__    
3 __NA__                3         42         6    __NA__    
4 __NA__                5         12         3    __NA__    
5 __NA__                5         23         1    __NA__    
6 __NA__                3         48         3    __NA__

In [40]:
colnames(data[,cols])

[1] "pid7"                  "birthyr"               "gender"               
 [4] "educ"                  "marstat"               "child18"              
 [7] "employ"                "employ_t"              "faminc_new"           
[10] "region"                "votereg"               "ideo5"                
[13] "newsint"               "presvote16post"        "presvote16post_t"     
[16] "pew_bornagain"         "pew_religimp"          "pew_churatd"          
[19] "pew_prayer"            "religpew"              "religpew_t"           
[22] "religpew_protestant"   "religpew_protestant_t" "immigrant"            
[25] "inputstate"            "race"                  "race_other"

In [43]:
head(data[, c("birthyr")])

[1] 1986 1972 1999 1975 1989 1992

In [42]:
head(2020 - data[, c("birthyr")])

[1] 34 48 21 45 31 28

For political party affiliation, there are quite a few variables associated with party - that needs investigation

house18, senate18, gov18, vote20dem, phecon, phhealth, phtax, phimmig, phborder, phforeign, phtrade, phopiod, phenviron, phnatdis, 

pid1d (Generally speaking, do you usually think of yourself as a Democrat, a Republican), 

pid1r (Generally speaking, do you usually think of yourself as a Republican, a Democrat),

pidstr (strength of affiliation)

pidlean (leaning towards which party?)